jerry-core/parser/js/js-parser-statm.c

```diff
@@ -1970,22 +2036,59 @@ parser_parse_statements (parser_context_t *context_p) /**< context */
         }
 
         lexer_next_token (context_p);
+
+#ifndef CONFIG_DISABLE_ES2015_CLASS
+        uint32_t constructor_super = PARSER_CLASS_CONSTRUCTOR | PARSER_CLASS_HAS_SUPER;
```

Comment:
Make this `const` so that it doesn't get modified by mistake, and move it before the switch so that it doesn't have to be defined multiple times (as around line 2194). Or make this something like a `#define PARSER_CLASS_CONSTRUCTOR_SUPER (PARSER_CLASS_CONSTRUCTOR | PARSER_CLASS_HAS_SUPER)` if the value is used multiple times, as in `js-parser.c`.

Reply:
`#define PARSER_CLASS_CONSTRUCTOR_SUPER (PARSER_CLASS_CONSTRUCTOR | PARSER_CLASS_HAS_SUPER)` is a good idea.